homeassistant/components/blink/config_flow.py

```diff
@@ -127,9 +127,10 @@ async def async_step_2fa(
         """Handle 2FA step."""
         errors = {}
         if user_input is not None:
-            pin: str = str(user_input.get(CONF_PIN))
             try:
-                valid_token = await _send_blink_2fa_pin(self.hass, self.auth, pin)
+                valid_token = await _send_blink_2fa_pin(
+                    self.hass, self.auth, str(user_input.get(CONF_PIN))
```

Comment:
Make `pin` optional in `_send_blink_2fa_pin`. Then we don't need to copy to string here.

Reply:
I think this is what you mean, my lack of expertise maybe showing :)